typescript/sdk/src/ism/HyperlaneIsmFactory.ts

```diff
@@ -0,0 +1,250 @@
+import { ethers } from 'ethers';
+
+import {
+  DomainRoutingIsm__factory,
+  IInterchainSecurityModule__factory,
+  StaticAggregationIsm__factory,
+  StaticMOfNAddressSetFactory,
+} from '@hyperlane-xyz/core';
+import { types } from '@hyperlane-xyz/utils';
+import { eqAddress } from '@hyperlane-xyz/utils/dist/src/utils';
+
+import { HyperlaneApp } from '../HyperlaneApp';
+import {
+  HyperlaneEnvironment,
+  hyperlaneEnvironments,
+} from '../consts/environments';
+import { HyperlaneContracts } from '../contracts';
+import { MultiProvider } from '../providers/MultiProvider';
+import { ChainMap, ChainName } from '../types';
+
+import { IsmFactoryFactories, ismFactoryFactories } from './contracts';
+import { DeployedIsm, IsmConfig, ModuleType } from './types';
+
+export class HyperlaneIsmFactory extends HyperlaneApp<IsmFactoryFactories> {
+  static fromEnvironment<Env extends HyperlaneEnvironment>(
+    env: Env,
+    multiProvider: MultiProvider,
+  ): HyperlaneIsmFactory {
+    const envAddresses = hyperlaneEnvironments[env];
+    if (!envAddresses) {
+      throw new Error(`No addresses found for ${env}`);
+    }
+    const fromAddressesMap = HyperlaneApp.fromAddressesMap(
+      envAddresses,
+      ismFactoryFactories,
+      multiProvider,
+    );
+    return new HyperlaneIsmFactory(
+      fromAddressesMap.contractsMap,
+      fromAddressesMap.multiProvider,
+    );
+  }
+
+  async deploy(chain: ChainName, config: IsmConfig): Promise<DeployedIsm> {
+    const signer = this.multiProvider.getSigner(chain);
+    switch (config.type) {
```

Comment:
Switch statements, double the nesting and half as intuitive! What's not to like! 😝 
But actually maybe consider breaking apart this deploy into separate methods for legibility?

Reply:
Tell me how you really feel